packages/bot-web-ui/src/stores/run-panel-store.js

```diff
@@ -259,15 +259,17 @@ export default class RunPanelStore {
         this.showClearStatDialog();
     }
 
-    clearStat() {
+    clearStat(is_websocket_disconnected = false) {
         const { summary_card, journal, transactions } = this.root_store;
 
         this.setIsRunning(false);
         this.setHasOpenContract(false);
         this.clear();
-        journal.clear();
-        summary_card.clear();
-        transactions.clear();
+        if (!is_websocket_disconnected) {
```

Comment:
@rupato-deriv Can we create a separate method to check the condition of the websocket connection and call the clear methods that you require in that. The clearStat is being used elsewhere and should not include the conditions in here.